src/Generator/Passes/CleanInvalidDeclNamesPass.cs

```diff
@@ -146,7 +146,14 @@ private static void Check(IEnumerable<Declaration> decls)
         {
             var anonymousDecls = decls.Where(p => string.IsNullOrEmpty(p.Name)).ToList();
             for (int i = 0; i < anonymousDecls.Count; i++)
-                anonymousDecls[i].Name = $"_{i}";
+            {
+                var anonymousDecl = anonymousDecls[i];
+
+                if (anonymousDecl.Namespace != null && anonymousDecl.Namespace.Name == anonymousDecl.Name)
```

Comment:
You could simplify to:
`anonymousDecl.Namespace?.Name == anonymousDecl.Name`

Reply:
I avoided that because I thought that if `anonymousDecl.Name` is null and `anonymousDecl.NameSpace` is null then it would be detected as a collision even tho there isn't one.